### PR TITLE
Authenticate with CIR using google ID token

### PR DESCRIPTION
**Pull request by berroar**
### What is the context of this PR?

Adds authentication for requests made to the CIR endpoint must to restrict and control API access. Reuses the same patterns that were added to support requests to SDS. A new environment variable has been added to support the change: `CIR_OAUTH2_CLIENT_ID`.

### How to review

Deploy the application and check that the logs to verify that the token has been generated. As CIR has not been integrated with our pipelines, we can only check that the token has been generated in the logs and that runner has then proceeded to the API call.

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by berroar** [comment]
[Runner: Authenticate with CIR using Google ID Token (2 SP)](https://trello.com/c/ur7kvwqD/5870-runner-authenticate-with-cir-using-google-id-token-2-sp)

**Review by petechd — APPROVED** [review]
Tested in pipelines, works as expected.

File: app/utilities/credentials.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import requests
+from flask import current_app
+
+from app.oidc.oidc import OIDCCredentialsService
+
+
+def fetch_credentials(session: requests.Session, client_id: str) -> None:
```
[the remark this answers is not in the document]
Comment (berroar) [reply app/utilities/credentials.py]: Renamed now 👍

**Review by rmccar — APPROVED** [review]

File: app/settings.py [hunk 2]
```diff
@@ -157,6 +157,7 @@ def utcoffset_or_fail(date_value, key):
 OIDC_TOKEN_LEEWAY_IN_SECONDS = int(os.getenv("OIDC_TOKEN_LEEWAY_IN_SECONDS", "300"))
 
 SDS_OAUTH2_CLIENT_ID = os.getenv("SDS_OAUTH2_CLIENT_ID")
+CIR_OAUTH2_CLIENT_ID = os.getenv("CIR_OAUTH2_CLIENT_ID")
```
[the remark this answers is not in the document]
Comment (MebinAbraham) [reply app/settings.py:160]: Added this and the missing SDS/OIDC ones.

**Review by petechd — APPROVED** [review]

**Pull request merged**